src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -0,0 +1,3670 @@
+    however, the original name is misleading as 'Drop Connect' is a different form of dropout in a separate paper...
+    See discussion: https://github.com/tensorflow/tpu/issues/494#issuecomment-532968956 ... I've opted for changing the
+    layer and argument names to 'drop path' rather than mix DropConnect as a layer name and use 'survival rate' as the
+    argument.
+    """
+    if drop_prob == 0.0 or not training:
+        return input
+    keep_prob = 1 - drop_prob
+    shape = (input.shape[0],) + (1,) * (input.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = keep_prob + torch.rand(shape, dtype=input.dtype, device=input.device)
+    random_tensor.floor_()  # binarize
+    output = input.div(keep_prob) * random_tensor
+    return output
+
+
+# Copied from transformers.models.beit.modeling_beit.BeitDropPath with Beit->GroundingDINO
+class GroundingDINODropPath(nn.Module):
+    """Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks)."""
+
+    def __init__(self, drop_prob: Optional[float] = None) -> None:
+        super().__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        return drop_path(hidden_states, self.drop_prob, self.training)
+
+    def extra_repr(self) -> str:
+        return "p={}".format(self.drop_prob)
+
+
+class GroundingDINOFusionLayer(nn.Module):
+    def __init__(self, config, init_values=1e-4):
+        super().__init__()
+        drop_path = config.fusion_droppath
+
+        # pre layer norm
+        self.layer_norm_vision = nn.LayerNorm(config.d_model)
+        self.layer_norm_text = nn.LayerNorm(config.d_model)
+        self.attn = GroundingDINOBiMultiHeadAttention(config)
+
+        # add layer scale for training stability
+        self.drop_path = GroundingDINODropPath(drop_path) if drop_path > 0.0 else nn.Identity()
+        self.vision_param = nn.Parameter(init_values * torch.ones((config.d_model)), requires_grad=True)
+        self.text_param = nn.Parameter(init_values * torch.ones((config.d_model)), requires_grad=True)
+
+    def forward(self, vision_features, text_features, attention_mask_vision=None, attention_mask_text=None):
+        vision_features = self.layer_norm_vision(vision_features)
+        text_features = self.layer_norm_text(text_features)
+        (delta_v, vision_attn), (delta_t, text_attn) = self.attn(
+            vision_features,
+            text_features,
+            vision_attention_mask=attention_mask_vision,
+            text_attention_mask=attention_mask_text,
+        )
+        vision_features = vision_features + self.drop_path(self.vision_param * delta_v)
+        text_features = text_features + self.drop_path(self.text_param * delta_t)
+
+        return (vision_features, vision_attn), (text_features, text_attn)
+
+
+# NOTE just renamed the class
+class GroundingDINODeformableLayer(nn.Module):
```

Comment:
copied from?

Reply:
It can be copied from `DeformableDetrEncoderLayer`, but I didn't find a way to copy from and maintain the name `GroundingDINODeformableLayer` since this module class is then used in `GroundingDINOEncoderLayer` is there a way to circumvent that issue with `Copied from`?

Reply:
Hmm indeed. Doesn't seem possible, cc @ydshieh